src/dsl.jl

```diff
@@ -298,149 +266,113 @@ end
 ### DSL Internal Master Function ###
 
 # Function for creating a ReactionSystem structure (used by the @reaction_network macro).
-function make_reaction_system(ex::Expr; name = :(gensym(:ReactionSystem)))
+function make_reaction_system(ex::Expr, name)
 
-    # Handle interpolation of variables
+    # Handle interpolation of variables in the input.
     ex = esc_dollars!(ex)
 
```

Comment:
This main function have now been reordered, trying to put the various option handling into separate sections depending on whether they deal wth declared symbols (must happen first), inferred symbols (happens after that) or other options (mostly must happen at the end). Previously stuff was a little bit all over the place depending on when it got added. Now it should be much clearer what is going on.